src/app/record/components/top-bar/modals/modal-name/modal-name.component.ts

```diff
@@ -0,0 +1,295 @@
+      familyName: namesForm.get('familyName').value,
+      creditName: namesForm.get('publishedName').value,
+      visibility: visibility,
+    } as NamesEndPoint
+  }
+
+  formToBackendOtherNames(namesForm: FormGroup): OtherNamesEndPoint {
+    const otherNames = {
+      errors: [],
+      otherNames: [],
+      visibility: this.originalBackendOtherNames.visibility,
+    }
+    this.otherNames.reverse()
+    this.otherNames
+      .map((value) => value.putCode)
+      .filter((key) => namesForm.value[key].otherName)
+      .forEach((key, i) => {
+        const otherName = namesForm.value[key].otherName
+        const visibility = namesForm.value[key].visibility
+        if (namesForm.value[key]) {
+          otherNames.otherNames.push({
+            putCode: key.indexOf('new-') === 0 ? null : key,
+            content: otherName,
+            displayIndex: i + 1,
+            source: this.userSession.userInfo.EFFECTIVE_USER_ORCID,
+            visibility: {
+              visibility,
+            },
+          } as Assertion)
+        }
+      })
+    return otherNames
+  }
+
+  saveEvent() {
+    this.loadingNames = true
+    this._recordNameService
+      .postNames(this.formToBackendNames(this.namesForm))
+      .subscribe(
+        (response) => {
+          console.log(response)
+          this._recordOtherNamesService
+            .postOtherNames(this.formToBackendOtherNames(this.namesForm))
+            .subscribe((res) => {
+              console.log(res)
+              this.closeEvent()
+            })
+        },
+        (error) => {
+          console.log(error)
+        }
+      )
+  }
+
+  closeEvent() {
+    this.dialogRef.close()
+  }
+
+  drop(event: CdkDragDrop<string[]>) {
+    moveItemInArray(this.otherNames, event.previousIndex, event.currentIndex)
+  }
+
+  addOtherName() {
+    this.namesForm.addControl(
+      'new-' + this.addedOtherNameCount,
+      new FormGroup({
+        otherName: new FormControl(),
+        visibility: new FormControl(this.defaultVisibility, {}),
+      })
+    )
+    this.otherNames.push({
+      putCode: 'new-' + this.addedOtherNameCount,
+      visibility: { visibility: this.defaultVisibility },
+    } as Assertion)
+    this.addedOtherNameCount++
+
+    this._changeDetectorRef.detectChanges()
+  }
+
+  deleteOtherName(putCode: string) {
+    const i = this.otherNames.findIndex((value) => value.putCode === putCode)
+    this.otherNames.splice(i, 1)
+    this.namesForm.removeControl(putCode)
+  }
+
+  getSourceName(names: Assertion) {
+    if (names.sourceName) {
+      if (names.lastModified) {
+        return (
+          names.sourceName +
+          ' ' +
+          names.lastModified.year +
+          '-' +
+          names.lastModified.month +
+          '-' +
+          names.lastModified.day
+        )
+      } else {
+        return names.sourceName
+      }
+    }
+  }
+
+  toGivenNames() {
+    document.getElementById('given-names').scrollIntoView()
```

Comment:
use the windows service to get the document. Like https://github.com/ORCID/orcid-angular/blob/129540b6d89ae718b92ab6ab0851188aeb822095/src/app/cdk/recaptcha/recaptcha.directive.ts#L110